packages/rich-text/src/concat.js

```diff
@@ -3,6 +3,24 @@
  */
 
 import { normaliseFormats } from './normalise-formats';
+import { create } from './create';
+
+/**
+ * Concats a pair of rich text values. Not that this mutates `a` and does NOT
+ * normalise formats!
+ *
+ * @param  {Object} a Value to mutate.
+ * @param  {Object} b Value to add read from.
+ *
+ * @return {Object} `a`, mutated.
+ */
+export function mergePair( a, b ) {
+	a.formats = a.formats.concat( b.formats );
```

Comment:
Unsure if it makes a practical impact as far as one being more performant than the other, but if you're mutating, you could just as well:

```js
a.formats.push( ...b.formats );
```

Reply:
> Unsure if it makes a practical impact as far as one being more performant than the other

788x more performant to use `Array#push`, if this is to be believed:

https://jsperf.com/javascript-array-concat-vs-push/4

Reply:
If we do that, each item that is not set will be set to `undefined`.

* `[ , , , ].concat( [ , , , ] )` gives you `[empty × 6]`
* `[ , , , ].push( ...[ , , , ] )` gives you `[empty × 3, undefined, undefined, undefined]`.

Not sure if there's another way.

Reply:
> If we do that, each item that is not set will be set to `undefined`.

TIL! Sparse arrays are very tricky, it seems 😄 

I don't see it being an issue. Just caught my attention as it seemed like it should behave the same. Since it doesn't, I see no reason to change.